flow/project.py

```diff
@@ -299,7 +303,7 @@ def keyfunction(job):
 
         def grouper(jobs):
             for key, group in groupby(sorted(jobs, key=keyfunction), key=keyfunction):
-                yield group
+                yield list(group)
```

Comment:
I think it would be much better to not rely on the fact that this is a list. Users might yield iterables as well. You can always create a list where you call the generator function.

Reply:
That's valid, I'll make this compatible.

Reply:
Also worth noting that you should only create a list, when you really need a list. There could be literally millions of jobs in that list which would take up a significant amount of memory.

Reply:
This means that there's also a need of optimization in my code because there are places where I created a list where it's not even necessary.

Reply:
Ok! Just be conscious of that, no need to optimize the code right now.